packages/color-modes/src/index.tsx

```diff
@@ -81,24 +81,17 @@ const useColorModeState = (theme: Theme = {}) => {
     return preferredMode || theme.initialColorModeName
   })
 
-  // on first render, we read the color mode from localStorage and
-  // clear the class on document element body
+  // remove initial color mode classes from the document and body
   useEffect(() => {
-    const stored = theme.useLocalStorage !== false && storage.get()
-
-    if (typeof document !== 'undefined') {
-      document.documentElement.classList.remove('theme-ui-' + stored)
-      document.body.classList.remove('theme-ui-' + stored)
-    }
-
-    if (stored && stored !== mode) {
-      mode = stored
-      setMode(stored)
+    const classMode = getModeFromClass()
```

Comment:
Perhaps it's faster to just attempt to remove classes without checking if they exist.